docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -11,18 +11,255 @@ Because the cloud and edge is located in the network boundary of different secur
+
+* Edgegateway is installed and run in the central cloud and each edge site respectively, so that these edge sites can access to the central cloud through edgegateway
+* Create edgesite objects to the system and associate edgesite with these edgegateway
+* All components in the cloud and edge that need mutual access are registered with the central cloud cloudgateway in the form of service, and the access policies of these services are configured
+* The central cloud synchronizes the policy to the corresponding edgegateway, which provides virtual presence for the edge through translation mechanisms to simulate local traffic flow
+
+### API Definition
+
+Using the architecture of arktos, the API is exposed through arktos API server, including service expose, policy management, site management, virtual presence management, etc
+
+![](Proposal-B3.png)
+
+#### API object
+
+Here are the API samples in arktos：
+
+	{
+	    "kind": "ESite",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "TokyoChibaSite1"
+	    }
+	}
+	
+	{
+	    "kind": "EGateway",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "TokyoChibaGateway"
+		},
+		"ip": "202.168.0.10",
+		"virtualPresenceIPcidr": "192.168.0.0/24",
+		"eSiteName": "TokyoChibaSite1"
+	}
+	
+	{
+	    "kind": "VirtualPresence",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "vp1"
+		},
+		"virtualIp": "192.168.0.10",
+		"eServiceName": "TykyoChibaNginx"
+	}
+	
+	{
+	    "kind": "EService",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "TykyoChibaNginx"
+		},
+		"port": 8080,
+		"protocol": "TCP",
+		"ip": "10.0.0.10",
+		"domainName": "EServiceNginx"
+		"eSiteName": "TokyoChibaSite1"
+	}
+	
+	{
+	    "kind": "EServer",
+		"apiVersion": "v1",
+		"eServerName": "Server1",
+		"ip": "10.0.0.20",
+		"eSiteName": "TokyoChibaSite1"
+	}
+	
+	{
+	    "kind": "EPolicy",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "Policy1"
+		},
+		"allowedServers": ["Server1"]
+	}
+	
+	{
+	    "kind": "ServiceExpose",
+	    "apiVersion": "v1",
+	    "eServiceName": "TykyoChibaNginx",
+		"dnsName": "url1.test.com",
+	    "eSites": ["TokyoChibaSite1"],
+	    "ePolicys": ["Policy1", "Policy2"]
+	}
+
+
+#### API Structs
+
+Here are the ESite data struct introduced into arktos，Represent each site, and the central cloud region also belongs to the site：
+
+	type ESite struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	}
+
+EGateway data struct：
+
+	type EGateway struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	
+		// Ip address of the gateway
+		Ip string
+	
+		// Virtual presence ip address cidr of this gateway
+		VirtualPresenceIPcidr string
+	
+		// ESiteName associated to the gateway
+		ESiteName string
+	}
```

Comment:
Have we considered the approach of defining all objects as CRDs?

Reply:
Yes, the idea now is to provide these objects and interfaces through the CRD method.